tools/ClassGen/InstrGen.cpp

```diff
@@ -624,6 +624,11 @@ int main(int argc, char **argv) {
 
 #include "Backends/CPU/CPUSpecificInstrs.h"
 #include "Backends/OpenCL/OpenCLSpecificInstrs.h"
+  // Add here external backend specific instructions headers.
```

Comment:
@opti-mix I like the idea, I think it would add complexity to this particular PR that needs to be handled carefully. I would then prefer to tackle such improvement in a separate PR after this one has landed.

The things that need to be taken care about with your proposal are the following:
- Today, the proposed external backend mechanism just knows about the fact that there is a ClassGen directory. It doesn't know whether there are backend specific nodes, specific instruction or both. Also, it doesn't know the name and location of the header files. Generating `ExternalBackendsSpecificInstrs.h` would mean that backends need to inform about their .h file and if they add specific nodes and/or specific instruction.
- We need to make sure that the dependencies are well managed. For instance, we don't want to recompile the glow library if there is no change in backend. It means that this `ExternalBackendsSpecificInstrs.h` should not be generated all the time and that the CMake build infrastructure should detect if there is any change in external backends before to touch this generated file.

Reply:
@tlepley-cadence Sound good! It is fine to tackle this improvement in a separate PR.